src/PrestaShopBundle/DataCollector/HookDataCollector.php

```diff
@@ -125,15 +125,26 @@ private function stringifyHookArguments(array &$hooksList)
     {
         foreach ($hooksList as &$hookList) {
             foreach ($hookList as &$hook) {
+                // Symfony Request object is quite heavy and is already given by debug tooling. Don't clone it again.
+                if (isset($hook['args']['request'])) {
+                    $hook['args']['request'] = 'Hook use client request as argument. For more details, cf. "Request/Response panel".';
+                }
+
                 $hook['args'] = $this->cloneVar($hook['args']);
 
                 foreach ($hook['modules'] as &$modulesByType) {
                     foreach ($modulesByType as $type => &$module) {
                         if (empty($module)) {
                             unset($modulesByType[$type]);
+                            continue;
                         }
 
                         if (array_key_exists('args', $module)) {
+                            // Symfony Request object is quite heavy and is already given by debug tooling. Don't clone it again.
+                            if (isset($module['args']['request'])) {
+                                $module['args']['request'] = 'Module use client request as argument. For more details, cf. "Request/Response panel".';
+                            }
+
```

Comment:
It doesn't look like this is related to the original bug, is it? 🤔

Reply:
@atomiix absolutely not 😅

It was a really weird bug, actually, the message for file size was indeed sent in the response, but after that the kernel called its `terminate` method, which goes through this HookCollector (which is used by the debug bar) and then it went out of memory.

When that happens there's an automatic error handler that appended another JSON message stating that memory went out of bound, but since it was appended after an initial JSON object the final result was unparsable.

```json
{
  error: 'File too big'
} {
  error: 'No more memory'
}
```

@MeKeyCool I think you need to describe all this in the PR's description because a reviewer can't guess it on first sight. And even for future dev who would track back the reason of this modification it's important to explain the context thoroughly in the description.

As to why it didn't happen previously, my guess is that the large file uses a big part of the memory in itself which means not much is left at the end of the process. So it doesn't mean this will never happen again, it just means the risks have been reduced.

Reply:
So the compromise we chose was to not display the Request in the Hokk parameters panel, the request is passed to almost all hooks, meaning the object is cloned several times in a process which is overkill since the debug bar already has a panel allowing to check the full Request with more details.

Reply:
So if I understand well, this bug only happens in debug mode, right?